### DISCUSSION TITLE
Total downloads of package via REST API

### DISCUSSION THREAD

**Discussion by omrozowicz-splunk** [community/community#35409] [Apps, API and Webhooks]
Hey, I'd like to gather the number of total downloads of packages for monitoring. The APIs are described here: https://docs.github.com/en/rest/packages, and the example outputs don't show any information about the total downloads.

But the number is being shown in github UI:
![image](https://user-images.githubusercontent.com/86965961/194512598-ab4cfd06-f363-4ea1-80ad-306f6e76f77c.png)
Is there any way to get that information via REST API?

The only close thing to that I found so far is `downloadsTotalCount` in [github's GraphQL API](https://docs.github.com/en/graphql/reference/objects#packageversionstatistics), but it doesn't work for me probably because of this being [deprecated](https://github.blog/changelog/2022-08-18-deprecation-notice-graphql-for-packages/) since August.

**Comment by jovezhong** [comment]
+1. Why there is no REST API for this? My current solution is to analyze the HTML DOM and extract the numbers. It's too much.

**Comment by fcavelti** [comment]
+1

**Comment by MasteraSnackin** [comment]
As of now, the GitHub REST API for Packages does not expose the aggregate “total downloads” count that you see in the UI, so you cannot retrieve that number directly via REST. The old GraphQL field downloadsTotalCount for PackageVersionStatistics was deprecated as part of the packages GraphQL deprecation and is not a viable option anymore.

At the moment the only workarounds are:

Scraping the HTML (which is brittle and not recommended), as others in this thread already noted.

Or, if you control distribution via releases rather than Packages, using the Releases REST API to sum download_count over all release assets, which is a different metric and only applies if you publish artefacts that way.
